Sources/Sentry/SentryTracer.m

```diff
@@ -492,30 +517,41 @@ - (void)finishInternal
     [SentryProfiler stopProfilingSpan:self.rootSpan];
 #endif // SENTRY_TARGET_PROFILING_SUPPORTED
 
-    SentryTransaction *transaction = [self toTransaction];
+    [self addMeasurementsAndChildrenToTransaction:_transaction];
 
-    // Prewarming can execute code up to viewDidLoad of a UIViewController, and keep the app in the
-    // background. This can lead to auto-generated transactions lasting for minutes or event hours.
-    // Therefore, we drop transactions lasting longer than SENTRY_AUTO_TRANSACTION_MAX_DURATION.
-    NSTimeInterval transactionDuration = [self.timestamp timeIntervalSinceDate:self.startTimestamp];
-    if ([self isAutoGeneratedTransaction]
-        && transactionDuration >= SENTRY_AUTO_TRANSACTION_MAX_DURATION) {
-        SENTRY_LOG_INFO(@"Auto generated transaction exceeded the max duration of %f seconds. Not "
-                        @"capturing transaction.",
-            SENTRY_AUTO_TRANSACTION_MAX_DURATION);
+    if ([self didAutomaticTraceExceedMaxDuration]) {
 #if SENTRY_TARGET_PROFILING_SUPPORTED
-        [SentryProfiler dropTransaction:transaction];
+        [SentryProfiler dropTransaction:_transaction];
 #endif // SENTRY_TARGET_PROFILING_SUPPORTED
         return;
     }
 
-    [_hub captureTransaction:transaction withScope:_hub.scope];
+    [_hub captureTransaction:_transaction withScope:_hub.scope];
 
 #if SENTRY_TARGET_PROFILING_SUPPORTED
-    [SentryProfiler linkTransaction:transaction];
+    [SentryProfiler linkTransaction:_transaction];
 #endif // SENTRY_TARGET_PROFILING_SUPPORTED
 }
 
+/**
+ * Prewarming can execute code up to viewDidLoad of a UIViewController, and keep the app in the
+ * background. This can lead to auto-generated transactions lasting for minutes or event hours.
+ * Therefore, we drop transactions lasting longer than SENTRY_AUTO_TRANSACTION_MAX_DURATION.
+ */
+- (BOOL)didAutomaticTraceExceedMaxDuration
```

Comment:
`SENTRY_AUTO_TRANSACTION_MAX_DURATION` can still result in transactions with incorrect durations that are several minutes long, can we not just detect prewarming using the environment variable and disable the auto UI transactions altogether in that case?

(This may not be in the scope of this PR, just an idea)

Reply:
This is not only about prewarming. It's a safety measure for all auto-generated transactions. Pre-warmed transactions are only one cause for such long-lasting transactions. @armcknight, can you maybe update the comment, please?

Reply:
@philipphofmann would you mind leaving a review comment with a revision for it?